### PR TITLE
fix: Use userInterestsConnection in collected artist rail and view

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [ONYX-155] 

### Description

This PR updates the collected artist rail and view to use `userInterestsConnection` instead of the deprecated `collectedArtistsConnection`. It also removes the artwork count in the list view.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Use userInterestsConnection in collected artist rail and view - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-155]: https://artsyproduct.atlassian.net/browse/ONYX-155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Components/MyCollectionBottomSheetModals/MyCollectionBottomSheetModals.tsx [hunk 1]
```diff
@@ -13,7 +13,8 @@ export const MyCollectionBottomSheetModals: React.FC<{}> = () => {
   const setViewKind = MyCollectionTabsStore.useStoreActions((actions) => actions.setViewKind)
   const view = MyCollectionTabsStore.useStoreState((state) => state.viewKind)
   const id = MyCollectionTabsStore.useStoreState((state) => state.id)
-  const uploadsCount = MyCollectionTabsStore.useStoreState((state) => state.artworksCount)
+  // TODO: Implement this
+  const uploadsCount = 0
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Components/MyCollectionBottomSheetModals/MyCollectionBottomSheetModals.tsx:17]: nice, we can remove this in the ticket after it

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistGridItem.tsx [hunk 2]
```diff
@@ -56,7 +56,6 @@ export const MyCollectionCollectedArtistGridItem: React.FC<
           setViewKind({
             viewKind: "Artist",
             id: artistData.internalID,
-            artworksCount: artworksCount,
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistGridItem.tsx:59]: it's nice to remove this, less coupling 🙌

**Review by MounirDhahri — APPROVED** [review]
🚀🚀🚀
Ole is on beast mode today

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsRail.tsx [hunk 3]
```diff
@@ -88,15 +88,15 @@ const collectedArtistsPaginationFragment = graphql`
   fragment MyCollectionCollectedArtistsRail_me on Me
   @argumentDefinitions(count: { type: "Int", defaultValue: 10 }, after: { type: "String" })
   @refetchable(queryName: "MyCollectionCollectedArtistsRail_myCollectionInfoRefetch") {
-    myCollectionInfo {
-      collectedArtistsConnection(
-        first: $count
-        after: $after
-        sort: TRENDING_DESC
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsRail.tsx:95]: no need to Ole yet. we can keep that for later.

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsRail.tsx:95]: if we can get the latest added ones first, that should be enough for us.

[the remark this answers is not in the document]
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsRail.tsx:95]: We could just set the default sort order to `-created_at`. I guess, there is no need to support custom sorting for now. I have opened a PR to set the default sort order.
: https://github.com/artsy/gravity/pull/16563

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsRail.tsx:95]: images

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx [hunk 4]
```diff
@@ -63,7 +63,7 @@ describe("MyCollectionCollectedArtistsView", () => {
   it("renders collected artist in a grid", async () => {
     const { getByText } = renderWithWrappers(<TestRenderer />)
     resolveMostRecentRelayOperation(mockEnvironment, {
-      MyCollectionInfo() {
+      Me() {
         return mockCollectedArtist
```
[the remark this answers is not in the document]
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx:67]: Good idea 👍 

https://github.com/artsy/eigen/pull/9026

**Pull request merged**